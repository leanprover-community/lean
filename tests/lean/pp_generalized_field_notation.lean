set_option pp.generalized_field_notation true

#check int.nat_abs_zero
#check int.sign_zero
#check int.sign_mul_nat_abs

lemma list.reverse_reverse {α} (xs : list α) : xs.reverse.reverse = xs :=
sorry

#check list.reverse_reverse

def prod.swap {α β} : α × β → β × α | ⟨a, b⟩ := ⟨b, a⟩

lemma prod.swap_swap {α β} (p : α × β) : p.swap.swap = p :=
by cases p; refl

#check prod.swap_swap