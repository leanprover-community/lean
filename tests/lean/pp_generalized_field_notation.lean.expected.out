int.nat_abs_zero : 0.nat_abs = 0
int.sign_zero : 0.sign = 0
int.sign_mul_nat_abs : ∀ (a : ℤ), a.sign * ↑(a.nat_abs) = a
pp_generalized_field_notation.lean:7:0: warning: declaration 'list.reverse_reverse' uses sorry
list.reverse_reverse : ∀ (xs : list ?M_1), xs.reverse.reverse = xs
prod.swap_swap : ∀ (p : ?M_1 × ?M_2), p.swap.swap = p
